### DISCUSSION TITLE
Mousepad and external USB mouse scroll in opposite directions

### DISCUSSION THREAD

**Discussion by remknul** [macOS-Dell-Latitude-E7470/macOS-Dell-Latitude-E7470#45] [Q&A]
I have got a problem with scrolling directions.

The mousepad on my Dell e7470 scrolls in the opposite direction as my externally connected USB mouse.
And as I frequently use both of them, this is very annoying.

Is there a way to revert the scroll direction in the driver software of the Dell mousepad?

Changing it in the MacOS system configuration does of course not work as it also changes the scroll direction of the external USB mouse and that would not solve my problem getting the behavior equal.

Thanks for any suggestions, Remco

**Comment by AlienKevin** [comment]
You may wanna take a look at [Scroll Reverser](https://pilotmoon.com/scrollreverser/). It's a free and open-source tool that allows you to control the scroll direction of the mouse and/or trackpads (you can select which one(s) the config applies to). I use it to reverse the direction of my trackpad and so far it works gorgeously.

**Reply by remknul** [reply]
Tanks for the suggestion, but the mousepad of the Dell e7470 is detected as generic mouse and not a trackpad by MacOS. (Sorry that I named it a trackpad in my initial message.)
The utility you referred to will, as a result of this, change the direction of both the mousepad-mouse and the USB external mouse at the same time.

Does somebody know if I can change the direction of the mousepad in the driver or by driver-parameters?

Is it correct that it is the VoodooPS2Controller.kext which is responsible for the Dell mousepad ?

**Reply by adityabakare** [reply]
> Tanks for the suggestion, but the mousepad of the Dell e7470 is detected as generic mouse and not a trackpad by MacOS. (Sorry that I named it a trackpad in my initial message.) The utility you referred to will, as a result of this, change the direction of both the mousepad-mouse and the USB external mouse at the same time.
> 
> Does somebody know if I can change the direction of the mousepad in the driver or by driver-parameters?
> 
> Is it correct that it is the VoodooPS2Controller.kext which is responsible for the Dell mousepad ?

Because we don't have proper trackpad drivers for our ALPS trackpad, the VoodooPS2Controller makes sure to make it fall back to the 'Generic Mouse' configuration.
By looking at the scroll reverser, it seems that it help to select different configs for both mouse and trackpad(although I haven't used it personally)

**Reply by remknul** [reply]
Oke, thanks for the info.

I will try to contact the developer(s) of the VoodoPS2Controller software and ask if it is possible to reverse the direction in their code. This to make the up-down behavior identical to an generic external USB mouse.

Regards, Remco
